saas/api/metrics.py

```diff
@@ -236,23 +236,24 @@ class RevenueMetricAPIView(ProviderMixin, APIView):
         }
     """
     def get(self, request, *args, **kwargs):
-        ends_at = datetime_or_now(request.GET.get('ends_at', None))
+        ends_at = request.GET.get('ends_at')
```

Comment:
We need to call `ends_at = datetime_or_now(request.GET.get('ends_at')` here because the parameters might be absent, in which case we want to default to now.
The reason to call datetime_or_now at the entry point of each function using a date parameter that can be `None` is to fix a consistent date. i.e. If we skip `datetime_or_now` then call two functions with `ends_at=None` both will run with a slight different `ends_at` date.